src/cljs/main/broadfcui/page/workspaces_list.cljs

```diff
@@ -91,23 +92,31 @@
    "No Access" #(= "NO ACCESS" (% "accessLevel"))
    })
 
-(def ^:private realm-types ["non-TCGA" "TCGA Open Access" "TCGA Protected Access"])
+(def ^:private realm-types ["Un-published" "Published" "TCGA Open Access" "TCGA Protected Access"])
+(def ^:private realm-types-defaults [true false false false])
 (def ^:private realm-predicates
- {"non-TCGA" #(not= (config/tcga-namespace) (get-in % ["workspace" "namespace"]))
+ {"Published" #(and (not= (config/tcga-namespace) (get-in % ["workspace" "namespace"])) (get-in % ["workspace" "attributes" "library:published"]))
+  "Un-published" #(and (not= (config/tcga-namespace) (get-in % ["workspace" "namespace"])) (not (get-in % ["workspace" "attributes" "library:published"])))
   "TCGA Open Access" #(and (= (config/tcga-namespace) (get-in % ["workspace" "namespace"]))
                            (not (get-in % ["workspace" "realm"])))
   "TCGA Protected Access" #(and (= (config/tcga-namespace) (get-in % ["workspace" "namespace"]))
                                 (get-in % ["workspace" "realm"]))
   })
 
 (def ^:private persistence-key "workspace-table-types")
+(def ^:private VERSION 1)
+
 
 (react/defc WorkspaceTable
   {:get-initial-state
    (fn []
-     (persistence/try-restore
-      {:key persistence-key
-       :initial (fn [] {:selected-types (reduce (fn [m x] (assoc m x true)) {} access-types)})}))
+     (let [persisted (persistence/try-restore
+                       {:key persistence-key
+                        :initial (fn [] {:v VERSION
+                                         :selected-types (merge (zipmap access-types access-types-defaults) (zipmap realm-types realm-types-defaults))})})]
+       (if (persistence/is-valid (comp (partial = VERSION) :v) persisted)
+         persisted
+         {:v VERSION :selected-types (merge (select-keys (:selected-types persisted) access-types) (zipmap realm-types realm-types-defaults))})))
```

Comment:
will check with moran to how important it is to keep the users selection for the access-control section of which workspaces to show